src/somd2/runner/_runner.py

```diff
@@ -185,19 +189,129 @@ def _check_space(self):
 
     def _check_directory(self):
         """
-        Check if the output directory has files already present.
-        Paired with the 'overwrite' option.
+        Find the name of the file from which simulations will be started.
+        Paired with the 'restart' option.
         """
-        import os as _os
-        import glob as _glob
+        from pathlib import Path as __Path
+        from ._dynamics import Dynamics
 
-        if self._config.overwrite:
-            return True
-        else:
-            if _glob.glob(_os.path.join(self._config.output_directory, "*.parquet")):
-                return False
+        fnames = {}
+        deleted = []
+        for lambda_value in self._lambda_values:
+            files = Dynamics.create_filenames(
+                self._lambda_values,
+                lambda_value,
+                self._config.output_directory,
+                self._config.restart,
+            )
+            fnames[lambda_value] = files
+            if not self._config.restart:
+                for file in files.values():
+                    fullpath = self._config.output_directory / file
+                    if __Path.exists(fullpath):
+                        deleted.append(fullpath)
+        if len(deleted) > 0:
+            if not self._config.supress_overwrite_warning:
+                _logger.warning(
+                    f"The following files already exist and will be overwritten: {list(set((deleted)))} \n"
+                )
+                input("Press Enter to erase and continue...")
+            # Loop over files to be deleted, ignoring duplicates
+            for file in list(set(deleted)):
+                file.unlink()
+        self._fnames = fnames
+
+    @staticmethod
+    def _compare_configs(config1, config2):
+        if not isinstance(config1, dict):
+            raise TypeError("'config1' must be of type 'dict'")
+        if not isinstance(config2, dict):
+            raise TypeError("'config2' must be of type 'dict'")
+
+        # Define the subset of settings that are allowed to change after restart
+        allowed_diffs = [
+            "runtime",
+            "restart",
+            "temperature",
+            "minimise",
+            "max_threads",
+            "equilibration_time",
+            "equilibration_timestep",
+            "energy_frequency",
+            "save_trajectory",
+            "frame_frequency",
+            "save_velocities",
+            "checkpoint_frequency",
+            "platform",
+            "max_threads",
+            "max_gpus",
+            "run_parallel",
+            "restart",
+            "save_trajectories",
+            "write_config",
+            "log_level",
+            "log_file",
+            "supress_overwrite_warning",
+            "xtra_args",
```

Comment:
Think this should be `extra_args`. However, we'd need to make sure that this doesn't modify anything that isn't allowed to change.

Reply:
This is a leftover from debugging that I forgot to remove. I'll remove `extra_args` from the options that are allowed to change as it is potentially dangerous - as it stands it's only modifying the `threads` argument, but it could be modified to change any option.

Reply:
It turns out this is slightly more complex than I originally thought - the `extra_args` dictionary needs to change due to the changeability `max_threads` argument. As it is not actually defined in the `config` itself, but rather added later, I'll make sure that it is skipped in the `to_dict` function and hide it.

Reply:
Ah yes, that's for things that need to be added via `map`. As you say, since it's not part of the actual config just hide it. The `max_threads` attribute will be checked by the config itself.